torch/_dynamo/guards.py

```diff
@@ -108,6 +108,10 @@ def uninteresting_files():
             "___skip_backend_check",
```

Comment:
Do we need similar `___skip_config_check` for run-only mode?

Reply:
might be able to handle run-only with `save_config=False`? No - this will still recompile under different config.

Reply:
No, you don't need this; the backend check skip is to deal with the fresh closure backend problem, IIRC. cc @ydwu4

Reply:
>  the fresh closure backend problem
We were not be able to solve it. Jason recommended to just re-compile incase of fresh closure problem.


> Do we need similar ___skip_config_check for run-only mode?

By design run-only mode will just re-use whatever that's cached most recently. It causes trouble for backend_match because it has a special backend that doesn't match with others. So we have to skip the backend guard check. In terms of configuration change, I guess we don't have to skip the config check as it won't be changed anyway. But I'm OK with short-circuiting the config check to be safer in case that config does change (we'll most likely end with a cache miss if config changes). But we might lose some minor perf due to this additional check for normal cases.